src/parser/grammar.ne

```diff
@@ -235,17 +268,20 @@ keyword ->
   ( %RESERVED_KEYWORD
   | %RESERVED_PHRASE
   | %RESERVED_JOIN
-  | %CASE
-  | %END
-  | %WHEN
-  | %ELSE
-  | %THEN
   | %AND
   | %OR
   | %XOR ) {%
   ([[token]]) => toKeywordNode(token)
 %}
 
+other_keyword ->
+  ( %WHEN
+  | %THEN
+  | %ELSE
+  | %END ) {%
+  ([[token]]) => toKeywordNode(token)
+%}
```

Comment:
This is to allow for these keywords to be used outside of CASE expressions, while only allowing them in specific locations inside a CASE expression. Otherwise we would get an ambiguous grammar.